client/src/components/Onboarding/OnboardingAdmin.js

```diff
@@ -1,7 +1,291 @@
+              View Log
+            </Button>
+          </div>
+        ))}
+      </td>
+    </tr>
+  );
+};
+
+OnboardingAdminListUser.propTypes = {
+  user: onboardingUserPropType.isRequired,
+  viewLogCallback: PropTypes.func.isRequired
+};
+
+const OnboardingAdminList = ({ users, viewLogCallback }) => {
+  return (
+    <table styleName="users">
+      <thead>
+        <tr>
+          <th>User</th>
+          <th>Step</th>
+          <th>Status</th>
+          <th>Administrative Actions</th>
+          <th>&nbsp;</th>
+          <th>Log</th>
+        </tr>
+      </thead>
+      <tbody>
+        {users.map(user => (
+          <OnboardingAdminListUser
+            user={user}
+            key={user.username}
+            viewLogCallback={viewLogCallback}
+          />
+        ))}
+      </tbody>
+    </table>
+  );
+};
+
+OnboardingAdminList.propTypes = {
+  users: PropTypes.arrayOf(onboardingUserPropType).isRequired,
+  viewLogCallback: PropTypes.func.isRequired
+};
+
+const OnboardingAdmin = () => {
+  const dispatch = useDispatch();
+  const [eventLogModalParams, setEventLogModalParams] = useState(null);
+
+  const { users, offset, limit, total, query, loading, error } = useSelector(
+    state => state.onboarding.admin
+  );
+
+  const paginationCallback = useCallback(
+    page => {
+      dispatch({
+        type: 'FETCH_ONBOARDING_ADMIN_LIST',
+        payload: {
+          offset: (page - 1) * limit,
+          limit,
+          query
+        }
+      });
+    },
+    [offset, limit, query]
+  );
+
+  const viewLogCallback = useCallback(
+    (user, step) => {
+      setEventLogModalParams({ user, step });
+    },
+    [setEventLogModalParams]
+  );
+
+  const toggleViewLogModal = useCallback(() => {
+    setEventLogModalParams();
+  }, [setEventLogModalParams]);
+
+  useEffect(() => {
+    dispatch({
+      type: 'FETCH_ONBOARDING_ADMIN_LIST',
+      payload: { offset, limit, query: null }
+    });
+  }, [dispatch, offset, limit]);
+
+  const current = Math.floor(offset / limit) + 1;
+  const pages = Math.ceil(total / limit);
+  if (loading) {
+    return <LoadingSpinner />;
+  }
+  if (error) {
+    return (
+      <Message type="warn">Unable to access Onboarding administration</Message>
+    );
+  }
+  return (
+    <div styleName="root">
+      <div styleName="container">
+        <div styleName="container-header">
+          <h5>Administrator Controls</h5>
+          <OnboardingAdminSearchbar />
+        </div>
+        {users.length === 0 && (
+          <div styleName="no-users-placeholder">
+            <Message type="warn">No users to show.</Message>
+          </div>
+        )}
+        <div styleName="user-container">
+          {users.length > 0 && (
+            <OnboardingAdminList
+              users={users}
+              viewLogCallback={viewLogCallback}
+            />
+          )}
+        </div>
+        {users.length > 0 && (
+          <div styleName="paginator-container">
+            <Paginator
```

Comment:
When invoking a new page action (previous, next, or index), it appears to sort of render twice - first a loading spinner, then one very brief hiccup of the rendered data followed by another loading spinner, then rendering stably

Reply:
Is this when you simulate many many users? How do I replicate?

Reply:
Yes, many users. I uncommented that line in the backend to respond with many copies of my user

Reply:
i'm not seeing the same behavior. chrome?

Reply:
Yep, chrome. I'll test again and take a video if I see it

Reply:
It's a very brief hiccup now. Looking in the network tab, the same request is being made twice

https://user-images.githubusercontent.com/20326896/110371745-01ed6080-8013-11eb-99ce-d6cb80bcdc59.mov

Reply:
ah that helps. ok i have some suspicions...

Reply:
Fixed